app.rb

```diff
@@ -0,0 +1,104 @@
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @person = []
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book) unless @books.include?(book)
+    puts 'Record is created successfully!!'
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Classroom: '
+    classroom = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has Parent Permission (true/false): '
+    has_permission = gets.chomp
+    student = Student.new(name, age.to_i, has_permission, classroom)
+    @person.push(student) unless @person.include?(student)
+    puts 'Record is created successfully!!'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    teacher = Teacher.new(name, age.to_i, nil, specialization)
+    @person.push(teacher) unless @person.include?(teacher)
+    puts 'Record is created successfully!! '
+  end
+
+  def list_books
+    if @books.length.zero?
+      puts 'No book found'
+    else
+      puts ''
+      @books.each_with_index do |book, index|
+        puts "#{index}) Title: #{book.title}   Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_persons
+    if @person.length.zero?
+      puts 'No person detail found'
+    else
+      puts ''
+      @person.each_with_index do |persons, index|
+        puts "#{index}) [Student] Name: #{persons.name} ID: #{persons.id} AGE: #{persons.age}" if persons.is_a?(Student)
+        puts "#{index}) [Teacher] Name: #{persons.name} ID: #{persons.id} AGE: #{persons.age}" if persons.is_a?(Teacher)
+      end
+    end
+  end
+
+  def list_rentals
+    if @rentals.length.zero?
+      puts 'No rental found for books'
+    else
+      puts ''
+      print 'ID of a person: '
+      id = gets.chomp
+      @rentals.each do |rental|
+        if rental.person.id.to_i == id.to_i
+          puts "Date: #{rental.date} Book: #{rental.book.title} by #{rental.book.author}"
```

Comment:
- [ ] Only the year of the rental date is been displayed. Please work on it to show the entire date entered by the user.

Your App Output |
:-------------------:|
<img width="253" alt="Screen Shot 2022-11-11 at 7 34 58 PM" src="https://user-images.githubusercontent.com/39523312/201417059-a2c12c8d-c6ee-46dd-821c-37dcfb65e32a.png"> |

Expected Output |
:-------------------:|
![Screen Shot 2022-11-11 at 7 37 22 PM](https://user-images.githubusercontent.com/39523312/201417138-ba25da4a-b93a-4298-9496-f610facb2635.png) |